app/domain/usecases/account/create_account.go

```diff
@@ -26,14 +28,14 @@ func (uc Usecase) Create(name, cpf, secret string) (account.Account, error) {
 
 	if err != nil {
 
-		return account.Account{}, ErrCreateAccount
+		return account.Account{}, fmt.Errorf("failed to create account instance: %w", err)
 	}
 
 	err = uc.Repository.Create(&newAccount)
 
 	if err != nil {
 
-		return account.Account{}, ErrCreateAccountRepository
+		return account.Account{}, fmt.Errorf("repository failed to create account: %w", err)
```

Comment:
Esse é um jeito que é muito usado mas, como você disse, acabando expondo detalhes da sua implementação pois repassa o err recebido.
Uma alternativa é inverter o que é passado no fmt.Errorf, passando o objeto de erro desta camada e o texto do erro recebido:

```suggestion
		return account.Account{}, fmt.Errorf("%w: %s", ErrCreateAccount, err.Error())
```

Reply:
Excelente!